src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/PCATangentNormalizationUtils.java

```diff
@@ -0,0 +1,305 @@
+package org.broadinstitute.hellbender.tools.pon.coverage.pca;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.primitives.Doubles;
+import org.apache.commons.math3.linear.DefaultRealMatrixChangingVisitor;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.stat.descriptive.rank.Median;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.mllib.linalg.DenseMatrix;
+import org.apache.spark.mllib.linalg.Matrix;
+import org.apache.spark.mllib.linalg.distributed.RowMatrix;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.exome.ReadCountCollection;
+import org.broadinstitute.hellbender.tools.pon.coverage.Case2PoNTargetMapper;
+import org.broadinstitute.hellbender.tools.pon.coverage.CoveragePanelOfNormals;
+import org.broadinstitute.hellbender.utils.GATKProtectedMathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.broadinstitute.hellbender.utils.spark.SparkConverter;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+/**
+ * Utility class for package-private methods for performing tangent normalization (and related operations).
+ *
+ * Currently, only supports tangent normalization in the reduced hyperplane, not the logNormal hyperplane
+ */
+public final class PCATangentNormalizationUtils {
+    private static final Logger logger = LogManager.getLogger(PCATangentNormalizationUtils.class);
+
+    /**
+     * Minimum target normalized and column centered count possible.
+     *
+     * <p>
+     *     It must be small yet greater than 0 to avoid -Inf problems in the calculations.
+     * </p>
+     */
+    public static double EPSILON = 1E-9;
+    private static double INV_LN2 = 1.0 / Math.log(2.0);
+    private static double LOG_2_EPSILON = Math.log(EPSILON) * INV_LN2;
+
+    private static final int TN_NUM_SLICES_SPARK = 50;
+
+    private PCATangentNormalizationUtils() {}
+
+    /**
+     * Target-factor normalizes a {@link RealMatrix} vector in-place given target factors..
+     */
+    static void factorNormalize(final RealMatrix input, final double[] targetFactors) {
+        Utils.nonNull(input, "Input matrix cannot be null.");
+        Utils.nonNull(targetFactors, "Target factors cannot be null.");
+        if (targetFactors.length != input.getRowDimension()) {
```

Comment:
I'm comfortable with throwing an `IllegalArgumentException` for this, which lets you use `Utils.validateArg`.